### PR TITLE
Add a getter to NavigatorState to return the current Route

### PR DESCRIPTION
**Pull request by xqwzts**
https://github.com/flutter/flutter/issues/14243 seemed like an easy first task, I added a getter to expose NavigatorState's `_history.last` property, and a related test to navigator_test.dart

### CODE REVIEW

File: packages/flutter/lib/src/widgets/navigator.dart [hunk 1]
```diff
@@ -834,6 +834,9 @@ class NavigatorState extends State<Navigator> with TickerProviderStateMixin {
     assert(() { _debugLocked = false; return true; }());
   }
 
+  /// The top-most [Route] in this navigator
+  Route<dynamic> get current => _history.last;
```
Comment (Hixie) [inline packages/flutter/lib/src/widgets/navigator.dart]: This should probably return null if _history is empty (currently this would throw).

**Comment by Hixie** [comment]
Before landing this, I'd like to make sure we have a clear use case. (We should probably have that discussion on the bug rather than here.)

This patch is great otherwise. Thanks in particular for the test.

File: packages/flutter/lib/src/widgets/navigator.dart (hunk 1 above)
Comment (xqwzts) [reply packages/flutter/lib/src/widgets/navigator.dart]: Done.

**Comment by xqwzts** [comment]
Thanks Hixie, I'll follow the discussion on the bug report, feel free to close this PR if it isn't needed. [I don't have a strong opinion on it myself, really only picked it up because it was rather easy to jump into]

**Comment by Hixie** [comment]
@xqwzts The patch itself is great.

File: packages/flutter/lib/src/widgets/navigator.dart [hunk 2]
```diff
@@ -834,6 +834,9 @@ class NavigatorState extends State<Navigator> with TickerProviderStateMixin {
     assert(() { _debugLocked = false; return true; }());
   }
 
+  /// The top-most [Route] in this navigator
```
Comment (Hixie) [inline packages/flutter/lib/src/widgets/navigator.dart]: one trivial nit: the documentation should end with a period.

Comment (xqwzts) [reply packages/flutter/lib/src/widgets/navigator.dart]: Done.

**Pull request closed without merging**